src/Uuid/Uuid.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+namespace Kirby\Uuid;
+
+use Closure;
+use Generator;
+use Hidehalo\Nanoid\Client as Nanoid;
+use InvalidArgumentException;
+use Kirby\Cms\Collection;
+use Kirby\Cms\File;
+use Kirby\Cms\Page;
+use Kirby\Cms\Site;
+use Kirby\Cms\User;
+use Kirby\Toolkit\Str;
+
+/**
+ * The `Uuid` classes provide an interface to connect
+ * indetifiable models (page, file, site, user, blocks,
+ * structure entries) with a dedicated UUID string.
+ * It also provides methods to cache these connections
+ * for faster lookup.
+ *
+ * ```
+ * // get UUID string
+ * $model->uuid()->render();
+ *
+ * // get model from an UUID string
+ * Uuid::for('page://HhX1YtRR2ImG6h4')->resolve();
+ *
+ * // cache actions
+ * $model->uuid()->populate();
+ * $model->uuid()->clear();
+ * ```
+ *
+ * @package   Kirby Uuid
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class Uuid
+{
+	protected const TYPE = 'uuid';
+
+	/**
+	 * Customisable callback function for generating new ID strings
+	 * instead of using Nanoid. Receives length of string as parameter.
+	 */
+	public static Closure|null $generator = null;
+
+	public Collection|null $context;
```

Comment:
Suggestion for a docblock to make clearer what the context is for and what it does:

```suggestion
	/**
	 * Collection that is likely to contain the model and
	 * that will be checked first to speed up the lookup
	 */
	public Collection|null $context;

```

I hope I understood it correctly.